e2e/Tests/provisioning/ProvisioningE2ETests.cs

```diff
@@ -707,7 +708,7 @@ public async Task DPS_Registration_AmqpWs_X509_IndividualEnrollment_InvalidGloba
                         VerboseTestLogger.WriteLine($"ProvisioningDeviceClient.RegisterAsync failed because: {ex.Message}");
                         await Task.Delay(TimeSpan.FromSeconds(2)).ConfigureAwait(false);
                     }
-                    catch (OperationCanceledException oce) when (overallCts.IsCancellationRequested)
+                    catch (OperationCanceledException oce) when (!overallCts.IsCancellationRequested)
```

Comment:
Previously, this would only catch TaskCanceledException/OperationCanceledException if the total retry timeout had passed. So the first timeout would not be caught, and the test would fail. Now this test will retry on a timeout up until the retry timeout of 3 minutes.